File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/request/RequestEntity.java
package com.roxiemobile.networkingapi.network.rest.request;

import com.roxiemobile.networkingapi.network.http.CookieStore;
import com.roxiemobile.networkingapi.network.http.HttpHeaders;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.net.URI;

public interface RequestEntity<T> {

    /**
     * TODO
     */
    @NotNull URI uri();

    /**
     * TODO
     */
    @Nullable HttpHeaders headers();

    /**
     * TODO
     */
    @Nullable CookieStore cookieStore();

    /**
     * TODO
     */
    @Nullable T body();
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/request/RequestEntity.kt
package com.roxiemobile.networkingapi.network.rest.request

import com.roxiemobile.networkingapi.network.http.CookieStore
import com.roxiemobile.networkingapi.network.http.HttpHeaders
import java.net.URI

interface RequestEntity<TBody>: Cloneable {

// MARK: - Properties

    /**
     * TODO
     */
    val link: URI

    /**
     * TODO
     */
    val httpHeaders: HttpHeaders

    /**
     * TODO
     */
    val cookieStore: CookieStore

    /**
     * TODO
     */
    val body: TBody?

// MARK: - Methods

    /**
     * TODO
     */
    public override fun clone(): RequestEntity<TBody>
}
